### ISSUE TITLE
Configuration.__getattr__ breaks some special attributes

### ISSUE THREAD

**Issue by cblegare** [ets-labs/python-dependency-injector#148]
Here is an example in a interactive python console:

```
>>> from dependency_injector.providers import Configuration
>>> c = Configuration('config')
>>> print(str(c.__module__))
<dependency_injector.providers.Configuration('config.__module__') at 0x7f4228bae3b8>
```

Expected behavior would be

```
>>> from dependency_injector.providers import Configuration
>>> c = Configuration('config')
>>> print(str(c.__module__))
dependency_injector.providers.Configuration
```

This breaks sphinx documentation build, for example

**Comment by rmk135** [comment]
Hello,

Yep, looks abnormal. I will work on it.

Thanks,
Roman

**Comment by rmk135** [comment]
Ok, so I think I see what's wrong here...

I think that expected behaviour should look like this:
```python
>>> object.__module__
'__builtin__'
>>> object().__module__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'object' object has no attribute '__module__'
```

It means that class should have `__module__` attribute, but instance of that class should not.

Going back to `Dependency Injector`, currently it behaves like this:

```python
>>> from dependency_injector import providers
>>>
>>> providers.Configuration.__module__
'dependency_injector.providers'
>>> providers.Configuration('config').__module__
<dependency_injector.providers.Configuration('config.__module__') at 0x103277a70>
```

The only problem I see is that `providers.Configuration('config').__module__` should raise an `AttributeError` for special attributes like `__module__`.

**Comment by rmk135** [comment]
Patch has been published in 3.3.6 release.

Finally, behaviour of providers is the same, like it is for `object`. Add example for demonstration:

```python
>>> from dependency_injector.providers import Configuration
>>>
>>> Configuration.__module__
'dependency_injector.providers'
>>> Configuration.__name__
'Configuration'
>>>
>>> Configuration('config').__module__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "src/dependency_injector/providers.pyx", line 660, in dependency_injector.providers.Configuration.__getattr__ (src/dependency_injector/providers.c:9423)
    raise AttributeError(
AttributeError: 'Configuration' object has no attribute '__module__'
>>> Configuration('config').__name__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "src/dependency_injector/providers.pyx", line 660, in dependency_injector.providers.Configuration.__getattr__ (src/dependency_injector/providers.c:9423)
    raise AttributeError(
AttributeError: 'Configuration' object has no attribute '__name__'
```

I really hope that this will not create any confusion for Sphinx because it seems to work correct on class level.

Thanks for raising this issue,
Hope it helps,
Roman

**Comment by cblegare** [comment]
That should indeed do the trick. Thank you very much!
